### PR TITLE
Rust - Double prefix fix for NewRecord #3032

### PR DESCRIPTION
**Pull request by alexswan10k**
Failing scenario + fix for #3032

### CODE REVIEW

**Comment by alexswan10k** [comment]
Actually this will work for a single usage I am not sure this is entirely correct. What we really want to be able to do is get the scoped alias that has been created (which may not be the same as the name).

I think there are a few places this pattern prevails so may well be worth checking all of them

**Comment by alexswan10k** [comment]
@ncave I may need some help. I have deliberately broken this again by adding a collision (local record + aliased record). We have this going on in a few places where we do this sort of thing, and assume the imported name is just fine.
```
let entNs, entName = splitNameSpace ent.FullName
```

What we need to do in all these cases is get the generated locally aliased name from the import. How do we do that?

The import correctly creates the alias by the way, so that bit works. It is just the transform of the actual NewRecord.

**Comment by ncave** [comment]
@alexswan10k See `getEntityFullName` it should give you a proper name when the entity is in another file.

**Comment by ncave** [comment]
@alexswan10k There may be other paces too, where we use `ent.FullName` directly, instead of passing it through `getEntityFullName`.

**Comment by alexswan10k** [comment]
I have just gone through them and I believe some of the usages are actually correct, as they are transforming the Class or Interface etc, so you really do want the raw value as there is no import yet.

I found 2 usages that were definitely wrong though and this seems to have fixed the problem.

Nice one. Just gotta wait for CI.

File: src/Fable.Transforms/Rust/Fable2Rust.fs [hunk 1]
```diff
@@ -3005,7 +3007,7 @@ module Util =
                 | Fable.DeclaredType(entRef, _) ->
                     let ent = com.GetEntity(entRef)
                     if ent.IsInterface then
-                        let entNs, entName = splitNameSpace ent.FullName
+                        let entName = getEntityFullName com ctx entRef
```
Comment (ncave) [inline src/Fable.Transforms/Rust/Fable2Rust.fs:3010]: @alexswan10k Do we need to split the full name here? Not saying we do, just asking.

Comment (alexswan10k) [reply src/Fable.Transforms/Rust/Fable2Rust.fs:3010]: I am not sure. Doesn't this just give you the imported alias name (or type name if no alias present)? We do have some test coverage for this area and everything passes. There could well be holes though.

**Pull request merged**